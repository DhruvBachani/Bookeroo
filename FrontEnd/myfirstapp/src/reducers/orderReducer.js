import { GET_USER_TRANSACTIONS, GET_ALL_TRANSACTIONS } from "../actions/types";

export const initialState = {
    userTransactions: [],
    allTransactions: []
};

export default function(state = initialState, action) {
    switch (action.type) {
        case GET_USER_TRANSACTIONS:
            return {
                ...state,
                userTransactions: action.payload
            };
        case GET_ALL_TRANSACTIONS:
            return {
                ...state,
                allTransactions: action.payload
            };

        default:
            return state;
    }
}
