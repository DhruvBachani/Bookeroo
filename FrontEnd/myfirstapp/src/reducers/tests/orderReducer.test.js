import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import { initialState } from '../orderReducer';

describe('orderReducer', () => {
    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    })

    it('should handle GET_USER_TRANSACTIONS', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_USER_TRANSACTIONS'
                })
        ).toMatchSnapshot();
    })

    it('should handle GET_ALL_TRANSACTIONS', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_ALL_TRANSACTIONS'
                })
        ).toMatchSnapshot();
    })

    it('should handle GET_ALL_BOOK_TRANSACTIONS', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_ALL_BOOK_TRANSACTIONS'
                })
        ).toMatchSnapshot();
    })

    it('should handle GET_CART_ITEMS', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_CART_ITEMS'
                })
        ).toMatchSnapshot();
    })

    it('should handle GET_CART_TOTAL', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_CART_TOTAL'
                })
        ).toMatchSnapshot();
    })
})