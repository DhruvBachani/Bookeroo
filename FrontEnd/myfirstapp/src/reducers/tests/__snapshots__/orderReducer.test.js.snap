// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderReducer should handle GET_ALL_BOOK_TRANSACTIONS 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": undefined,
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`orderReducer should handle GET_ALL_TRANSACTIONS 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": undefined,
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`orderReducer should handle GET_CART_ITEMS 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": undefined,
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`orderReducer should handle GET_CART_TOTAL 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": undefined,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`orderReducer should handle GET_USER_TRANSACTIONS 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": undefined,
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;

exports[`orderReducer should return the initial state 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "order": Object {
    "allTransactions": Array [],
    "cartItems": Array [],
    "cartTotal": 0,
    "purchasedBooks": Array [],
    "userTransactions": Array [],
  },
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
  "user": Object {
    "allUsers": Array [],
    "userRequests": Array [],
  },
}
`;
